cfgen/cfgen

```diff
@@ -1073,16 +1077,19 @@ class ConfProcess(ToDhall):
 
         facts = node_desc['facts']
         proto = node_desc.get('data_iface_type', 'tcp')
-        # Creates Lnet endpoint, remove once deprecated.
-        # ep = Endpoint(proto=NetProtocol[proto],
-        #               ipaddr=facts[ipaddr_key(node_desc['data_iface'])],
-        #              portal=proc_t.value)
-        ep = LibfabricEndpoint(NetFamily='inet', proto=NetProtocol[proto],
-                               ipaddr=facts[ipaddr_key(
-                                            node_desc['data_iface'])],
-                               portalgroup=proc_t.value)
+        transport_type = node_desc.get('transport_type', 'libfab')
+        ep: Any = (
+            LibfabricEndpoint(NetFamily='inet', proto=NetProtocol[proto],
```

Comment:
Consider this as a generic endpoint,
Transport_type can also be of type "sock" which too follows the same format as that of libfabric.

Reply:
Yeah a new endpoint format can be added easily, just need to add it to the union and add a corresponding rendering function. Can do this once we enable sock transport.

Reply:
Having specific static typing helps to implement strict typing during configuration and avoid any unwanted and dynamic changes to the formats.